packages/tailwindcss/src/utilities.ts

```diff
@@ -2785,6 +2792,699 @@ export function createUtilities(theme: Theme) {
+    staticUtility('mask-clip-view', [['mask-clip', 'view-box']])
+    staticUtility('mask-no-clip', [['mask-clip', 'no-clip']])
+
+    /**
+     * @css `mask-origin`
+     */
+
+    staticUtility('mask-origin-border', [['mask-origin', 'border-box']])
+    staticUtility('mask-origin-padding', [['mask-origin', 'padding-box']])
+    staticUtility('mask-origin-content', [['mask-origin', 'content-box']])
+    staticUtility('mask-origin-fill', [['mask-origin', 'fill-box']])
+    staticUtility('mask-origin-stroke', [['mask-origin', 'stroke-box']])
+    staticUtility('mask-origin-view', [['mask-origin', 'view-box']])
+
+    /**
+     * @css `mask-image`
+     */
+
+    staticUtility('mask-none', [['mask-image', 'none']])
+
+    let maskPropertiesGradient = () =>
+      atRoot([
+        property('--tw-mask-linear', 'linear-gradient(#fff, #fff)'),
+        property('--tw-mask-radial', 'linear-gradient(#fff, #fff)'),
+        property('--tw-mask-conic', 'linear-gradient(#fff, #fff)'),
+      ])
+
+    type MaskStopDescription = {
+      color: (value: string) => AstNode[] | undefined
+      position: (value: string) => AstNode[] | undefined
+    }
+
+    function maskStopUtility(classRoot: string, desc: MaskStopDescription) {
+      utilities.functional(classRoot, (candidate) => {
+        if (!candidate.value) return
+
+        // Arbitrary values
+        if (candidate.value.kind === 'arbitrary') {
+          let value: string | null = candidate.value.value
+          let type =
+            candidate.value.dataType ?? inferDataType(value, ['length', 'percentage', 'color'])
+
+          switch (type) {
+            case 'color': {
+              value = asColor(value, candidate.modifier, theme)
+              if (value === null) return
+
+              return desc.color(value)
+            }
+            case 'percentage': {
+              if (candidate.modifier) return
+              if (!isPositiveInteger(value.slice(0, -1))) return
+
+              return desc.position(value)
+            }
+            default: {
+              if (candidate.modifier) return
+
+              return desc.position(value)
+            }
+          }
+        }
+
+        // Known values: Color Stops
+        {
+          let value = resolveThemeColor(candidate, theme, ['--background-color', '--color'])
+          if (value) {
+            return desc.color(value)
+          }
+        }
+
+        // Known values: Positions
+        {
+          if (candidate.modifier) return
+
+          let type = inferDataType(candidate.value.value, ['number', 'percentage'])
+          if (!type) return
+
+          switch (type) {
+            case 'number': {
+              let multiplier = theme.resolve(null, ['--spacing'])
+              if (!multiplier) return
+              if (!isValidSpacingMultiplier(candidate.value.value)) return
+
+              return desc.position(`--spacing(${candidate.value.value})`)
```

Comment:
Oh! First time we use `--spacing(…)` in our own utilities. This does mean we need an additional step, but I think we do that anyway so no big deal.

We could do some cleanup and apply this for other utilities as well, but that's for a future improvement.

Reply:
Oh heh yeah I can change this if we want/need

Reply:
I changed this to inline the implementation